syntax = "proto3";

message Mensaje{
    string id = 1; // id unica mensaje
    string mensaje = 2; // mensaje
    float timestamp = 3;  // marca de tiempo
    string usernameEmisor = 4;  // username del emisor
    string usernameReceptor = 5;  // username del receptor
}

message MensajeReply{
    int32 value = 1; // 1 si el mensaje se entrego correctamente, de lo contrario 2
}

message Cliente{
    string username = 1; // nombre de usuario 
}

message Vacio{}

service Chat{
    rpc ChatStream(Vacio) returns (stream Mensaje){}
    
    rpc AgregarCliente(Cliente) returns (MensajeReply) {}  // al conectarse un cliente, se agrega a la lista del servidor

    rpc EnviarMensaje(Mensaje) returns (MensajeReply) {}

    rpc ListadoClientes(Vacio) returns (stream Cliente) {}

    rpc MensajesEnviadosPor(Cliente) returns (stream Mensaje) {}
}


// python -m grpc_tools.protoc -I --python_out=. --grpc_python_out=. chat.proto
